libsrc/wrf_io.F90

```diff
@@ -1,307 +1,319 @@
-!*  hereby granted, provided that this disclaimer notice appears in 
-!*  all copies.  All modifications to this software must be clearly
-!*  documented, and are solely the responsibility of the agent making
-!*  the modification.  If significant modifications or enhancements
-!*  are made to this software, the SMS Development team
-!*  (sms-info@fsl.noaa.gov) should be notified.
+!* Standard Disclaimer
+!*
+!* Forecast Systems Laboratory
+!* NOAA/OAR/ERL/FSL
+!* 325 Broadway
+!* Boulder, CO 80303
+!*
+!* AVIATION DIVISION
+!* ADVANCED COMPUTING BRANCH
+!* SMS/NNT Version: 2.0.0
+!*
+!* This software and its documentation are in the public domain and
+!* are furnished "as is". The United States government, its
+!* instrumentalities, officers, employees, and agents make no
+!* warranty, express or implied, as to the usefulness of the software
+!* and documentation for any purpose. They assume no
+!* responsibility (1) for the use of the software and documentation;
+!* or (2) to provide technical support to users.
+!*
+!* Permission to use, copy, modify, and distribute this software is
+!* hereby granted, provided that this disclaimer notice appears in
+!* all copies. All modifications to this software must be clearly
+!* documented, and are solely the responsibility of the agent making
+!* the modification. If significant modifications or enhancements
+!* are made to this software, the SMS Development team
+!* (sms-info@fsl.noaa.gov) should be notified.
+!*
+!*----------------------------------------------------------------------------
+!*
+!* WRF NetCDF I/O
+! Author: Jacques Middlecoff jacquesm@fsl.noaa.gov
+!* Date: October 6, 2000
 !*
 !*----------------------------------------------------------------------------
 
 module wrf_data
 
-  integer                , parameter      :: FATAL            = 1
-  integer                , parameter      :: WARN             = 1
-  integer                , parameter      :: WrfDataHandleMax = 99
-  integer                , parameter      :: MaxDims          = 2000 ! = NF_MAX_VARS
-#if(WRF_CHEM == 1)
-  integer                , parameter      :: MaxVars          = 10000
-#else
-  integer                , parameter      :: MaxVars          = 3000
-#endif
-  integer                , parameter      :: MaxTimes         = 10000
-  integer                , parameter      :: DateStrLen       = 19
-  integer                , parameter      :: VarNameLen       = 31
-  integer                , parameter      :: NO_DIM           = 0
-  integer                , parameter      :: NVarDims         = 4
-  integer                , parameter      :: NMDVarDims       = 2
-  character (8)          , parameter      :: NO_NAME          = 'NULL'
-  character (DateStrLen) , parameter      :: ZeroDate = '0000-00-00-00:00:00'
-
-#include "wrf_io_flags.h"
-
-  character (256)                         :: msg
-  logical                                 :: WrfIOnotInitialized = .true.
-
+  integer , parameter :: FATAL = 1
```

Comment:
We  need test code for this.

Reply:
The build problems not finding netcdf on Jet were resolved by communication from Kyle.  I had missed a github module checkout to support gsi cmake.   It now builds on Jet and emc-post builds using this library.   It also builds
on wcoss-cray and all regression tests survive execution but results have not been compared with a baseline yet.  I need
to regenerate the baseline but this test is expected to pass and the remaining requested changes are from Ed Hartnett

Reply:
Roger that, I'll cleanup the code and make the changes @edwardhartnett suggested now that we know it works.

Reply:
All post numbers compare for the post regression tests on wcoss-cray so this new libray has not broken anything in emc-post and I am confident of it's functionality.  Edward's concerns about testing of the code for robustness remain and are unaddressed by passing of regression tests

Reply:
Well we do what we can do. ;-) We can't do everything we would like, unfortunately.